examples/type_system/lib/common_fixes_analysis.dart

```diff
@@ -131,3 +131,17 @@ void funcCast() {
   filterValues((x) => (x as String).contains('Hello'));
   // #enddocregion func-cast
 }
+
+//-----------------------------------------------
+
+// #docregion type-inf-null
+  List<int> ints = [1, 2, 3];
+  var maximumOrNull = ints.fold(null,
+      (a, b) => a == null || a < b ? b : a);
+// #enddocregion type-inf-null
```

Comment:
You should put these in methods as they both can't be in the global scope as their names will collide.

You also need to ignore the resulting errors of the failed type inference since they will cause code analysis to fail.

We currently enforce [`dart format`](https://dart.dev/tools/dart-format) being run on files, so they will end up looking like this too:

```suggestion
void infNull() {
  // #docregion type-inf-null
  List<int> ints = [1, 2, 3];
  // ignore: non_bool_operand, undefined_operator
  var maximumOrNull = ints.fold(null, (a, b) => a == null || a < b ? b : a);
  // #enddocregion type-inf-null    
} 
```